src/main/kotlin/RailNetwork.kt

```diff
@@ -0,0 +1,72 @@
+
+class RailNetwork(segments:List<RailSegment>, trains:List<Train>){
+
+    private var simulationStep:Int=1
+    private var segments:List<RailSegment> = segments
+    private var trains:List<Train> = trains
+
+    fun simulateOneStep(){
+
+        for(train:Train in trains){
+
+            var currentSegment:Int = train.getCurrentSegment(simulationStep)
+            if(currentSegment > 0) {
+
+                segments[currentSegment - 1].addTrain(train) // -1 because Segment 1 in the Schedule is here Segment 0 in the Array.
+
+            }
+
+        }
+
+        for(segment:RailSegment in segments){
+
+            var delayed = false
```

Comment:
Die Bestimmung ob ein Segment ausgelastet ist, kannst du in `RailSegment` verschieben, z.B. als `checkDelay` oder so. Damit kannst du die dortigen Methoden `getCapacity`, `getTrainCount`, und `setTrainDelayed`.